addons/payment/static/src/js/post_processing.js

```diff
@@ -95,9 +95,28 @@ odoo.define('payment.post_processing', function (require) {
                 }
                 return nbTx;
             }
-            // if there's only one tx to manage
+                       
+            /*
+            * When the server sends the list of monitored transactions, it tries to post-process 
+            * all the successful ones. If it succeeds or if the post-process has already been made, 
+            * the transaction is removed from the list of monitored transactions and won't be 
+            * included in the next response. We assume that successful and post-process 
+            * transactions should always prevail on others, regardless of their number or state.
+            */
+            if (render_values['tx_done'].length === 1 &&
+                render_values['tx_done'][0].is_post_processed) {
+                    if(render_values['tx_done'][0].landing_route){
+                        window.location = render_values['tx_done'][0].landing_route;
+                        return;
+                    }
```

Comment:
Per the API, a transaction should always have a `landing_route`. Why do you need to check it before redirecting?

```suggestion
                    window.location = render_values['tx_done'][0].landing_route;
                    return;
```

Reply:
@chevalierv Thanks I wasn't aware I'll remove the check
V15 was already merged I made a new PR: https://github.com/odoo/odoo/pull/101126

Reply:
@chevalierv done 
thanks for the feedback